agent/rpc/peering/service.go

```diff
@@ -379,34 +370,49 @@ func (s *Server) PeeringList(ctx context.Context, req *pbpeering.PeeringListRequ
 	// reconcile the actual peering state; need to copy over the ds for peering
 	var cPeerings []*pbpeering.Peering
 	for _, p := range peerings {
-		cp := copyPeeringWithNewState(p, s.reconciledStreamStateHint(p.ID, p.State))
-
-		// add imported services count
-		st, found := s.Tracker.StreamStatus(p.ID)
-		if !found {
-			s.Logger.Trace("did not find peer in stream tracker when listing peers", "peerID", p.ID)
-		} else {
-			cp.ImportedServiceCount = uint64(len(st.ImportedServices))
-			cp.ExportedServiceCount = uint64(len(st.ExportedServices))
-		}
-
+		cp := s.reconcilePeering(p)
 		cPeerings = append(cPeerings, cp)
 	}
+
 	return &pbpeering.PeeringListResponse{Peerings: cPeerings}, nil
 }
 
-// TODO(peering): Maybe get rid of this when actually monitoring the stream health
-// reconciledStreamStateHint peaks into the streamTracker and determines whether a peering should be marked
-// as PeeringState.Active or not
-func (s *Server) reconciledStreamStateHint(pID string, pState pbpeering.PeeringState) pbpeering.PeeringState {
-	streamState, found := s.Tracker.StreamStatus(pID)
+// TODO(peering): Get rid of this func when we stop using the stream tracker for imported/ exported services and the peering state
+// reconcilePeering enriches the peering with the following information:
+// -- PeeringState.Active if the peering is active
+// -- ImportedServicesCount and ExportedServicesCount
+// NOTE: we return a new peering with this additional data
+func (s *Server) reconcilePeering(peering *pbpeering.Peering) *pbpeering.Peering {
+	streamState, found := s.Tracker.StreamStatus(peering.ID)
+	if !found {
+		s.Logger.Trace("did not find peer in stream tracker; cannot populate imported and"+
+			" exported services count or reconcile peering state", "peerID", peering.ID)
+		return peering
```

Comment:
Nice! We even have copy-on-write in the case where we aren't tracking the stream.

Reply:
Should this be a WARN? It looks like a bad state we'd want to detect